lib/haml/railtie.rb

```diff
@@ -10,13 +10,32 @@
 module Haml
   class Railtie < ::Rails::Railtie
     initializer :haml do |app|
-      require "haml/template"
-      if defined?(::Sass::Rails::SassTemplate) && app.config.assets.enabled
-        require "haml/sass_rails_filter"
+      if defined?(ActiveSupport)
```

Comment:
I just left the checks in place, but agree. Not sure why we need to check for that given this is a railtie and the whole file shouldn't be loaded unless we are in rails. It's also in place before I touched this code and in haml 5.0.0.beta2